Service/Request/RequestFactory.php

```diff
@@ -173,13 +173,21 @@ private function validateEndpointPath(string $path): bool
      */
     private function parseQueryParams(string $path): array
     {
+        $queryParamsArray = [];
         $queryParamsString = parse_url($path, PHP_URL_QUERY);
         if (null !== $queryParamsString) {
+            // $queryParamsArray: ['query-param' => '{queryParam}']
```

Comment:
good morning. to be honest I prefer to see here a story WHY do we need this, with input examples and output results. And consider use strategy here and keep this code in a separate class with dedicated unit tests

Reply:
The story is pretty simple - we have a bug that query parameters aren't encoded for all GET requests (for example, `/v1/car-types/built-dates?country=CH&main-type=Qashqai+2&sort=&manufacturer=225` - `Qashqai+2` should be `Qashqai%2B2`). 
And this very change I made because for a placeholder in the URI we use query param value, not the key.

Reply:
please add this as a comment to the method

Reply:
ok, thanks, makes sense

Reply:
These comments look excessive and distracting. The function description is enough.